homeassistant/components/bluetooth/models.py

```diff
@@ -91,6 +91,10 @@ class BaseHaScanner:
     def discovered_devices(self) -> list[BLEDevice]:
         """Return a list of discovered devices."""
 
+    @abstractmethod
+    async def async_get_device_by_address(self, address: str) -> BLEDevice | None:
```

Comment:
Ideally in the future bleak supports this natively and we can make this a callback instead.